spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -425,6 +455,21 @@ private static void importUnpartitionedSparkTable(SparkSession spark, TableIdent
       List<DataFile> files = TableMigrationUtil.listPartition(
           partition, Util.uriToString(sourceTable.location()), format.get(), spec, conf, metricsConfig, nameMapping);
 
+      if (checkDuplicateFiles) {
+        Dataset<Row> fileDs = spark.createDataset(
+            Lists.transform(files, f -> f.path().toString()), Encoders.STRING())
+            .toDF("file_path");
+
+        Dataset<Row> existingFiles = loadMetadataTable(spark, targetTable, MetadataTableType.ENTRIES);
+        Column joinCond = existingFiles.col("data_file.file_path").equalTo(fileDs.col("file_path"));
+        Dataset<Row> duplicates = fileDs.join(existingFiles, joinCond);
+        List<String> duplicateList = Arrays.stream((Row[]) duplicates.head(10))
```

Comment:
could do takeAsList to make this a bit simpler also if we push the selection into the join it will be a cheaper join

So two things here

duplicates = importedFiles.join(existingFiles, joinCond).select("file_path").as(Encoders.String)
list<String> duplicateList = duplicates.takeAsList(10)

Reply:
Yea that was my first version, I hit #2783 as I mentioned above (spent a good half day on it and couldn't find any easy fix)

Reply:
oh yeah .. :( the fix I think would be to select a column to prevent the pruning issue ... which would be hacky. I guess we should really fix that pruning issue :(

Reply:
I still think it would be easier to do .takeAsList(10) instead of the array conversion but nbd there

Reply:
Whoops missed that part of the comment.  You are right, done.

Reply:
Yea and before it was doing 'duplicates.select("file_path").head(10)' and still failed, I think the pruning issue got worse in 3.1 (wrote some test in #2783)  (Or was there another select you had in mind?)

Reply:
Finally changed this, after you merged fix for #2783